### PR TITLE
bugfix - intermittent channel handshake failure

### LINKED ISSUE #926: Feature request: print out tx content if failed to estimate gas
Currently, if rly fails to estimate gas before broadcasting a tx, it will print out something like this to the console:

```
2022-08-22T00:33:48.111796Z	info	Error building or broadcasting transaction	{"provider_type": "cosmos", "chain_id": "wasm-test-1", "attempt": 1, "max_attempts": 5, "error": "rpc error: code = InvalidArgument desc = invalid connection hop ID: identifier cannot be blank: invalid identifier: invalid request"}
```

Which contains the error trace, but not the tx itself, making it hard to debug. It would be helpful if the tx body can be printed out here as well.

(btw, if anyone knows what could possibly cause "hop ID" to be empty when creating a channel, please also let me know! it happens sometime when registering new interchain accounts, but not always.)

### PR DESCRIPTION
**Pull request by agouin**
Resolves a caching issue where the counterparty connection ID will be empty if a `MsgConnectionOpenInit` message was observed in the history after the initial connection state was cached by the `CosmosChainProcessor`. This was causing the following failure of the `MsgConnectionOpenTry` tx:
```
ts=2022-08-23T05:21:03.068934Z lvl=info msg="Error building or broadcasting transaction" provider_type=cosmos chain_id=icad-2 attempt=1 max_attempts=5 error="rpc error: code = Unknown desc = invalid connection hop ID: identifier cannot be blank: invalid identifier: unknown request"
```

Fixes #926

### CODE REVIEW

**Review by mark-rushakoff — COMMENTED** [review]
The code changes look good, but is there a test you can add to ensure we retain this behavior?

**Comment by DavidNix** [comment]
> The code changes look good, but is there a test you can add to ensure we retain this behavior?

+1 A unit test(s) will minimize regressions in the future.

**Comment by jtieri** [comment]
Additionally, #926 specifically requests for a tx to be logged in the case of some error occurring when attempting to broadcast a tx and then OP mentions this empty connection hop ID bug as well in the comments. This would definitely fix the bug mentioned in the issue comments but it doesn't necessarily solve the main issue described in the title.

File: relayer/chains/cosmos/message_handlers_test.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+	conntypes "github.com/cosmos/ibc-go/v5/modules/core/03-connection/types"
+	chantypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	"github.com/cosmos/relayer/v2/relayer/processor"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+func TestConnectionStateCache(t *testing.T) {
```
Comment (jtieri) [inline relayer/chains/cosmos/message_handlers_test.go:14]: 🔥 🔥 🔥

File: relayer/processor/types_internal.go [hunk 2]
```diff
@@ -196,6 +198,38 @@ type outgoingMessages struct {
 	chanMsgs []channelMessageToTrack
 }
 
+// MarshalLogObject satisfies the zapcore.ObjectMarshaler interface
+// so that you can use zap.Object("messages", r) when logging.
+// This is typically useful when logging details about a partially sent result.
+func (om *outgoingMessages) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	for i, m := range om.pktMsgs {
+		pfx := "pkt_" + strconv.FormatInt(int64(i), 10) + "_"
+		enc.AddString(pfx+"evt_type", m.msg.eventType)
```
Comment (jtieri) [inline relayer/processor/types_internal.go]: Should these read `event_type`?

**Review by jtieri — APPROVED** [review]
This looks good Andrew, really good comments and unit tests

File: relayer/processor/types_internal.go (hunk 2 above)
Comment (agouin) [reply relayer/processor/types_internal.go]: I tried to keep the log keys as slim as possible, but happy to expand for clarity

**Pull request merged**